03-astropy-overview/io_fits_ascii_overview.ipynb

```diff
@@ -432,37 +316,53 @@
    "metadata": {},
    "outputs": [],
    "source": [
-    "from astropy.table import QTable\n",
-    "from astropy.utils.data import download_file"
+    "mask_hdu.header"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "table_filename = download_file('http://data.astropy.org/tutorials/FITS-tables/chandra_events.fits', cache=True)"
+    "You might expect that you can use the `.data` attribute to access the data, but that does not work:"
    ]
   },
   {
    "cell_type": "code",
    "execution_count": null,
-   "metadata": {},
+   "metadata": {
+    "editable": true,
+    "slideshow": {
+     "slide_type": ""
+    },
+    "tags": [
+     "raises-exception"
+    ]
+   },
    "outputs": [],
    "source": [
-    "# When a table contains columns with units, QTable can be used\n",
-    "# instead of Table.\n",
-    "tab = QTable.read(table_filename, hdu=1)"
+    "mask_hdu.data"
    ]
   },
   {
-   "cell_type": "code",
-   "execution_count": null,
+   "cell_type": "markdown",
    "metadata": {},
-   "outputs": [],
    "source": [
-    "tab.colnames"
+    "#### Explanation\n",
+    "\n",
+    "What just happened?! It certainly appeared as if the mask HDU had been loaded since we could access its header, but the data clearly was not loaded. \n",
+    "\n",
+    "The error message, `alueError: I/O operation on closed file`, provides a clue.\n",
+    "\n",
+    "The reason is primarily historical. FITS files are, by default, opened in *memory map* mode to avoid reading in the entire file at once. In memory map mode, the `.data` that looks like an array is actually, because of the way numpy handles memory maps, a refernce to the file with the data. That has been the default in astropy since its very beginning, including in the stand-alone `pyfits` package that preceded, and then became a part of, astropy. \n",
```

Comment:
```suggestion
    "The reason is primarily historical. FITS files are, by default, opened in *memory map* mode to avoid reading in the entire file at once. In memory map mode, the `.data` that looks like an array is actually, because of the way numpy handles memory maps, a reference to the file with the data. That has been the default in astropy since its very beginning, including in the stand-alone `pyfits` package that preceded, and then became a part of, astropy. \n",
```